### PR TITLE
Production build

### PR DESCRIPTION
**Pull request by ckcollab**

### CODE REVIEW

File: src/utils/storage.py [hunk 1]
```diff
@@ -11,6 +12,10 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
 
+class WhiteNoiseStaticFilesStorage(CompressedManifestStaticFilesStorage):
```
Comment (jimmykodes) [inline src/utils/storage.py]: maybe a comment about why we need this?

File: src/wsgi.py [hunk 2]
```diff
@@ -1,8 +1,7 @@
 import os
 
 from django.core.wsgi import get_wsgi_application
-from whitenoise.django import DjangoWhiteNoise
 
-settings_module = os.environ.setdefault('DJANGO_SETTINGS_MODULE', "settings.base")
+settings_module = os.environ.setdefault('DJANGO_SETTINGS_MODULE', "settings.develop")
 
-application = DjangoWhiteNoise(get_wsgi_application())
+application = get_wsgi_application()
```
Comment (jimmykodes) [inline src/wsgi.py]: Where does whitenoise stuff get declared now? Did I miss it?

File: circle.yml [hunk 3]
```diff
@@ -51,7 +51,7 @@ jobs:
           command: docker-compose -f docker-compose.yml -f docker-compose.selenium.yml exec django py.test src/tests/functional/ -m e2e
           no_output_timeout: 60m
 
-      - run: docker-compose exec django flake8 src/ --ignore E501
+      - run: docker-compose exec django flake8 src/ --ignore E501,F405
```
[the remark this answers is not in the document]
Comment (ckcollab) [reply circle.yml]: It's raised in quite a few places is why I did this I believe:
```
$ docker-compose exec django flake8 src/ --ignore E501
src/settings/develop.py:6:1: F405 'INSTALLED_APPS' may be undefined, or defined from star imports: settings.base
src/settings/develop.py:8:1: F405 'MIDDLEWARE' may be undefined, or defined from star imports: settings.base
src/settings/production.py:4:17: F405 'os' may be undefined, or defined from star imports: .base
src/settings/production.py:12:17: F405 'os' may be undefined, or defined from star imports: .base
src/settings/production.py:13:14: F405 'os' may be undefined, or defined from star imports: .base
src/settings/production.py:14:19: F405 'os' may be undefined, or defined from star imports: .base
src/settings/production.py:15:23: F405 'os' may be undefined, or defined from star imports: .base
src/settings/production.py:16:14: F405 'os' may be undefined, or defined from star imports: .base
src/settings/production.py:17:17: F405 'os' may be undefined, or defined from star imports: .base
src/settings/production.py:18:22: F405 'os' may be undefined, or defined from star imports: .base
src/settings/production.py:19:16: F405 'os' may be undefined, or defined from star imports: .base
```

File: src/apps/commands/management/commands/upload_backup.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
+from os import remove
+from os.path import join
```
[the remark this answers is not in the document]
Comment (ckcollab) [reply src/apps/commands/management/commands/upload_backup.py]: old old old stuff, refactored to `import os`

File: src/apps/commands/management/commands/upload_backup.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+class Command(BaseCommand):
+    help = "Takes a database dump file and puts it on remote storage"
+
+    def add_arguments(self, parser):
+        parser.add_argument('backup_path', type=str, help='Path to backup to upload, relative to /app/backups/')
+
+    def handle(self, *args, **options):
+        backup_file_name = options['backup_path']
+        backup_path = join("/app/backups", options['backup_path'])
```
[the remark this answers is not in the document]
Comment (ckcollab) [reply src/apps/commands/management/commands/upload_backup.py]: Yeah I think that's an OK compromise for now, every instance running this runs it in docker. We could read this path from `CODALAB_BACKUP_PATH` or something to make it more extendable.. hmm... think it's worth it?

File: src/settings/production.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+from .base import *  # noqa: F401,F403
```
[the remark this answers is not in the document]
Comment (ckcollab) [reply src/settings/production.py]: 👍

File: src/templates/base.html [hunk 7]
```diff
@@ -124,36 +124,35 @@
...
             </div>
             <div class="three wide column">
-                <h4 class="ui inverted header">Services</h4>
-                <div class="ui inverted link list">
-                    <a href="#" class="item">Banana Pre-Order</a>
-                    <a href="#" class="item">DNA FAQ</a>
-                    <a href="#" class="item">How To Access</a>
-                    <a href="#" class="item">Favorite X-Men</a>
-                </div>
+{#                <h4 class="ui inverted header">Services</h4>#}
```
[the remark this answers is not in the document]
Comment (ckcollab) [reply src/templates/base.html]: Replaced with `{# Empty for now, can add a list of links here if needed! #}`

File: src/templates/base.html [hunk 8]
```diff
@@ -124,36 +124,35 @@
...
+{#                    <a href="#" class="item">DNA FAQ</a>#}
+{#                    <a href="#" class="item">How To Access</a>#}
+{#                    <a href="#" class="item">Favorite X-Men</a>#}
+{#                </div>#}
             </div>
             <div class="seven wide column">
-                <h4 class="ui inverted header">Footer Header</h4>
-                <p>Extra space for a call to action inside the footer that could help re-engage users.</p>
+                <h4 class="ui inverted header">Codalab v2</h4>
+                <p>Join us on <a href="https://github.com/codalab/competitions-v2">Github</a> for contact & bug reports</p>
```
[the remark this answers is not in the document]
Comment (ckcollab) [reply src/templates/base.html]: I think it doesn't matter but better safe than sorry :)

File: src/templates/base.html [hunk 9]
```diff
@@ -206,7 +205,7 @@ <h4 class="ui inverted header">Footer Header</h4>
             },
             API: "/api/",
             STATIC: function (file_name) {
-                return "{% static '!-to-be-replaced-!' %}".replace('!-to-be-replaced-!', file_name)
+                return "{{ STATIC_URL }}" + file_name
```
[the remark this answers is not in the document]
Comment (ckcollab) [reply src/templates/base.html]: 👍

File: src/utils/storage.py (hunk 1 above)
Comment (ckcollab) [reply src/utils/storage.py]: just pulling since we're not even using this

File: src/wsgi.py (hunk 2 above)
Comment (ckcollab) [reply src/wsgi.py]: Whitenoise is used in development only, via:

```python
# settings/develop.py

MIDDLEWARE += ('whitenoise.middleware.WhiteNoiseMiddleware',)
```

File: circle.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jimmykodes) [reply circle.yml]: 👍

File: src/wsgi.py (hunk 2 above)
Comment (jimmykodes) [reply src/wsgi.py]: yea, but don't we still have to wrap our wsgi app in whitenoise stuff? not just activate the middleware?

File: src/apps/commands/management/commands/upload_backup.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (jimmykodes) [reply src/apps/commands/management/commands/upload_backup.py]: Probably fine, yea

File: src/wsgi.py (hunk 2 above)
Comment (ckcollab) [reply src/wsgi.py]: http://whitenoise.evans.io/en/stable/django.html#enable-whitenoise

No I don't think we need to do WSGI stuff for Django, just middleware is fine.

File: src/templates/base.html (hunk 8 above)
[the remark this answers is not in the document]
Comment (jimmykodes) [reply src/templates/base.html]: 👍

**Pull request merged**